src/calc-length.js

```diff
@@ -93,6 +93,72 @@
     return new CalcLength(calcDictionary);
   };
 
+  CalcLength.prototype._addCalcLengths = function(addedLength) {
+    if (!(addedLength instanceof CalcLength)) {
+      throw new TypeError('Objects not of type CalcLength');
+    } 
+    
+    var calcDictionary = {};
+
+    //Iterate through all possible length types and add there values
+    for(var i = 0; i < shared.LengthValue.LengthType.length; i++){
+      var type = shared.LengthValue.LengthType[i];
+      if (this[type] == null){
```

Comment:
It's clearer to (find the things that are defined and add them) than it is to (find the things that are undefined and ad the other thing).

if (addedLength[type] != null && this[type] != null) {
calcDictionary[type] = this[type] + addedLength[type];
} else if (this[type] != null) {
calcDictionary[type] = this[type];
} else if (addedLength[type] != null) {
calcDictionary[type] = addedLength[type];
}